sia-ant/job_balancemaintainer.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"log"
+	"time"
+
+	"github.com/NebulousLabs/Sia/api"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// balanceMaintainer mines currency. If the wallet balance is above desiredBalance,
+// the miner will be stopped, otherwise the miner will run until the
+// desiredBalance is reached.
+func (j *JobRunner) balanceMaintainer(desiredBalance types.Currency) {
+	j.tg.Add()
+	defer j.tg.Done()
+
+	if err := j.client.Get("/miner/start", nil); err != nil {
+		log.Printf("[%v balanceMaintainer ERROR]: %v\n", j.siaDirectory, err)
+		return
+	}
+	// Every 20 seconds, check the balance. If it's over the desired balance,
+	// start the miner. Otherwise, stop the miner.
+	for {
+		select {
+		case <-j.tg.StopChan():
+			return
+		case <-time.After(time.Second * 20):
+		}
+
+		var walletInfo api.WalletGET
+		err := j.client.Get("/wallet", &walletInfo)
+		if err != nil {
+			log.Printf("[%v balanceMaintainer ERROR]: %v\n", j.siaDirectory, err)
+		}
+		if walletInfo.ConfirmedSiacoinBalance.Cmp(desiredBalance) < 0 {
+			log.Printf("[%v balanceMaintainer INFO]: not enough currency, starting the miner\n", j.siaDirectory)
```

Comment:
this will be printed every 20s until enough currency is mined. Maybe check the current miner state and do nothing if it is already mining?